engine/runtime-integration-tests/src/test/scala/org/enso/interpreter/test/instrument/InstrumentTestContext.scala

```diff
@@ -130,7 +130,13 @@ abstract class InstrumentTestContext(packageName: String) {
       }
       .flatten
       .filter(excludeLibraryLoadingPayload)
-      .toList ++ lastSeen
+      .toList
+
+    if (lastSeen.isEmpty || lastSeen.get == collected.last) {
+      collected
+    } else {
+      collected ++ lastSeen
+    }
```

Comment:
@4e6, there was a bug in this method resulting in duplication of the last message. That problem was very visible in my workflow:
- call `receiveN(1000, 5)` - e.g. receive all one can with small timeout
- verify how much one got - `N`
- modify to `receiveN(N, 60)`

This _test writing workflow_ did result in a 60s timeout. With 7268fc98190b8918e56593aa0ceaac34eb034a9a change it will work.